### PR TITLE
Global cache rename

### PR DESCRIPTION
**Pull request by markscheel**
## Proposed changes

Renames ConstGlobalCache to GlobalCache.

This looks like a large PR, but should be easy to review commit-by-commit.  Most of the changed lines are from the
first commit that just does a `git mv` of several files.  The next commit does a global-replace ConstGlobalCache->GlobalCache, another commit does 'const_global_cache' -> 'global_cache' (except for some places where the 'const' will remain), and then other commits make Clang-tidy happy because this PR touches many files.

This is done in anticipation of changes that add another TaggedTuple to GlobalCache that has non-const items (like FunctionsOfTime).

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [x] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: docs/DevGuide/ParallelExecutable/MinimalExecutable.md [hunk 1]
```diff
@@ -298,18 +298,18 @@ takes the command line options as an argument (as type
 - Parses the input file options, populating a tagged tuple with their
   values.  (This is discussed in more detail in a future tutorial on
   input file options.)
-- Creates the `ConstGlobalCache` (a nodegroup chare) that holds
+- Creates the `GlobalCache` (a nodegroup chare) that holds
```
Comment (kidder) [inline docs/DevGuide/ParallelExecutable/MinimalExecutable.md:301]: another doc place mentioning constant

File: docs/DevGuide/ParallelExecutable/Concepts.md [hunk 2]
```diff
@@ -184,7 +184,7 @@ essentially a one-element array.
 
 SpECTRE does not use the readonly global variables provided by
 Charm++.  Instead SpECTRE provides a nodegroup called the
-`ConstGlobalCache` which provides global access to read-only objects,
+`GlobalCache` which provides global access to read-only objects,
```
Comment (kidder) [inline docs/DevGuide/ParallelExecutable/Concepts.md]: not `read-only` anymore

**Review by kidder — CHANGES_REQUESTED** [review]
fails to compile (looks like from something merged earlier today)

File: tests/Unit/Parallel/Test_AlgorithmCore.cpp [hunk 3]
```diff
@@ -720,7 +720,6 @@ struct TestMetavariables {
       case Phase::AnyOrderStart:
         return Phase::AnyOrderFinish;
       case Phase::AnyOrderFinish:
```
Comment (kidder) [inline tests/Unit/Parallel/Test_AlgorithmCore.cpp:722]: I think there is a `[[fallthrough]]` attribute to mark that this is intentional...

**Comment by markscheel** [comment]
Rebased on develop, added fixup commits to make it compile again and addressing PR comments.

**Review by nilsdeppe — COMMENTED** [review]
LGTM please go ahead and squash

**Review by kidder — COMMENTED** [review]
yes squash

**Comment by markscheel** [comment]
Thanks!  Squashed.

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**